src/util/asset_analysis.py

```diff
@@ -0,0 +1,63 @@
+"""Helper functions used to analyze data on assets."""
+
+from packaging.version import InvalidVersion, Version
+
+
+def is_older_than_version(
+        supported: list[str],
+        target_version: Version) -> bool:
+    """Determines if an asset that supports a list of versions is older than a
```

Comment:
I'm starting to think we should disable the commenting rule for the linter. This is a lot of excessive commenting that adds very little value to such a straightforward function.